app/javascript/react/store/measurementsSlice.ts

```diff
@@ -0,0 +1,112 @@
+import { createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { AxiosResponse } from "axios";
+import { oldApiClient } from "../api/apiClient";
+import { API_ENDPOINTS } from "../api/apiEndpoints";
+import { Error, StatusEnum } from "../types/api";
+import { getErrorMessage } from "../utils/getErrorMessage";
+
+interface Measurement {
+  time: number;
+  value: number;
+  latitude: number;
+  longitude: number;
+}
+
+export interface MeasurementsState {
+  data: Measurement[];
+  minMeasurementValue: number;
+  maxMeasurementValue: number;
+  averageMeasurementValue: number;
+  status: StatusEnum;
+  error?: Error;
+  isLoading: boolean;
+}
+
+const initialState: MeasurementsState = {
+  data: [],
+  minMeasurementValue: 0,
+  maxMeasurementValue: 0,
+  averageMeasurementValue: 0,
+  status: StatusEnum.Idle,
+  isLoading: false,
```

Comment:
you will need to add error: null to initial state, so we know there is no error on init